community/contributing/pr_review_guidelines.rst

```diff
@@ -0,0 +1,357 @@
+.. _doc_pr_review_guidelines:
+
+Pull request review process
+===========================
+
+.. note::
+
+    This page is intended to provide insight into the pull request (PR) review
+    process that we aspire to. As such, it is primarily targeted at engine
+    maintainers who are responsible for reviewing and approving pull requests.
+    That being said, much of the content is useful for prospective contributors
+    wanting to know how to ensure that their PR is merged.
+
+From a high level, the ideal life cycle of a pull request looks like: 
+  1. A contributor opens a PR that fixes a specific problem.
```

Comment:
"..(optimally closing Github issue or implementing a proposal)", could also link to more info about the proposal process?